account_credit_control/models/credit_control_policy.py

```diff
@@ -0,0 +1,389 @@
+        lines = self._due_move_lines(controlling_date)
+        to_add, to_remove = self._get_partner_related_lines(controlling_date)
+        lines = (lines | to_add) - to_remove
+        to_add, to_remove = self._get_invoice_related_lines(controlling_date)
+        lines = (lines | to_add) - to_remove
+        return lines
+
+    @api.multi
+    @api.returns('account.move.line')
+    def _lines_different_policy(self, lines):
+        """ Return a set of move lines ids for which there is an
+            existing credit line but with a different policy.
+        """
+        self.ensure_one()
+        different_lines = self.env['account.move.line']
+        if not lines:
+            return different_lines
+        cr = self.env.cr
+        cr.execute("SELECT move_line_id FROM credit_control_line"
+                   "    WHERE policy_id != %s and move_line_id in %s"
+                   "    AND manually_overridden IS false",
+                   (self.id, tuple(lines.ids)))
+        res = cr.fetchall()
+        if res:
+            return different_lines.browse([row[0] for row in res])
+        return different_lines
+
+    @api.multi
+    def check_policy_against_account(self, account):
+        """ Ensure that the policy corresponds to account relation """
+        allowed = self.search(['|', ('account_ids', 'in', account.ids),
+                               ('do_nothing', '=', True)])
+        if self not in allowed:
+            allowed_names = "\n".join(x.name for x in allowed)
+            raise UserError(
+                _('You can only use a policy set on '
+                  'account %s.\n'
+                  'Please choose one of the following '
+                  'policies:\n %s') % (account.name, allowed_names)
+            )
+        return True
+
+
+class CreditControlPolicyLevel(models.Model):
+    """Define a policy level. A level allows to determine if
+    a move line is due and the level of overdue of the line"""
+
+    _name = "credit.control.policy.level"
+    _order = 'level'
+    _description = """A credit control policy level"""
+
+    name = fields.Char(required=True, translate=True)
+    policy_id = fields.Many2one('credit.control.policy',
+                                string='Related Policy',
+                                required=True)
+    level = fields.Integer(required=True)
+    computation_mode = fields.Selection(
+        [('net_days', 'Due Date'),
+         ('end_of_month', 'Due Date, End Of Month'),
+         ('previous_date', 'Previous Reminder')],
+        string='Compute Mode',
+        required=True
+    )
+    delay_days = fields.Integer(string='Delay (in days)', required=True)
+    email_template_id = fields.Many2one('mail.template',
+                                        string='Email Template',
+                                        required=True)
+    channel = fields.Selection([('letter', 'Letter'),
+                                ('email', 'Email')],
```

Comment:
Selection items are repeated twice, in attribute `channel` in model `credit.control.line` and here in model `credit.control.policy.level` model.

I think we must create a list and reference it here to avoid to have to change this values twice if we increase values in future.